pyrobuf/parse_proto.py

```diff
@@ -209,17 +222,19 @@ def _parse_message(self, s, current, tokens, enums, imported_enums):
 
         for token in tokens:
             if token.token_type == 'MESSAGE':
-                current.messages[token.name] = self._parse_message(s, token, tokens, enums, imported_enums)
+                token.full_name = current.full_name + token.name
+                current.messages[token.name] = self._parse_message(s, token, tokens, messages, enums, imported_enums)
+                # updates the dictionary of known/parsed messages.
+                messages.update(current.messages)
```

Comment:
A few things here:

1. It would be slightly more efficient to just use `messages[token.name] = current.messages[token.name]` here rather than copying every k:v pair each time.
2. This breaks scoping rules somewhat. By mutating this global dictionary, it's now possible for outer scoped messages to use the child messages of their siblings. This approach also means you can't have two messages, `A` and `B`, that both have distinct children named `C`, even though that's perfectly legal according to formal protobuf spec.
3. Related to above, in the above example we should be able to reference `A.C` in message `B`.

That being said, I think this is a great start and gets us most of the way there.